### PR TITLE
WD-11655 - create /microcloud

### PR DESCRIPTION
**Pull request by lizzochek**
## Done

Created /microcloud page. Also updated vanilla version to the latest to use new components. Updated run script to fix error originating because of one of the pip packages removal. 

## QA

- [demo link](https://jp-ubuntu-com-485.demos.haus/microcloud)
- [copy doc](https://docs.google.com/document/d/1KzG5qScbehgotJYYWBDM82-9BnUN7mc4Omg9Sa2Nehg/edit)
- [figma](https://www.figma.com/design/tI8YK4wQnKiWLAxdX1fWnB/24.10-jp.ubuntu.com?node-id=342-7303&t=MXVTKOd1rVLI7Oa7-0)
- View the site on demo
- Run through the following [QA steps](https://discourse.canonical.com/t/qa-steps/152)
- Check that the page is correct according to figma and copy doc

## Issue / Card
[WD-11655](https://warthogs.atlassian.net/browse/WD-11655)

[WD-11655]: https://warthogs.atlassian.net/browse/WD-11655?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by lyubomir-popov** [comment]
- pls replace the image in the header with this one: https://assets.ubuntu.com/v1/43d1d899-Canonical-MicroCloud.png
- add the highlgihted background, and set the aspect ratio to 16:9 as usual (use the new vanilla p-image classes for both)
- remove the suru underneath
- remove this as it is a duplicate of the new top image:
image

**Comment by immortalcodes** [comment]
I think you would also need to change the navigation, currently microcloud in nav bar redirects to ubuntu/microcloud

**Comment by immortalcodes** [comment]
Are we fine with these images on small screen with edges touching to lines?
image

**Comment by immortalcodes** [comment]
> Are we fine with these images on small screen with edges touching to lines? image

I think we can remove images on small screen and fix them to not touch lines on small and medium screens

File: templates/microcloud.html [hunk 1]
```diff
@@ -0,0 +1,249 @@
+{% extends "_base/base.html" %}
+{% set active_page = "microcloud" %}
+
+
```
Comment (immortalcodes) [inline templates/microcloud.html:4]: Extra blank line

File: templates/microcloud.html [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+<section class="p-section">
+    <div class="u-fixed-width p-section--shallow">
+        <hr />
+        <h2>短時間でクラウドの運用を開始</h2>
+    </div>
+    <div class="row--25-75-on-large">
+        <div class="col">
+            <div class="p-equal-height-row has-divider-1">
+                <div class="p-equal-height-row__col">
+                    <div class="p-equal-height-row__item">
```
Comment (immortalcodes) [inline templates/microcloud.html:55]: Here in we can use `u-hide-small` to hide the image and not disturb other formats

File: templates/microcloud.html [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+<section class="p-section">
+    <div class="row--50-50">
+        <hr />
+        <div class="col">
+            <h2>MicroCloudのサイズ</h2>
+        </div>
+        <div class="col p-section--shallow">
+            <div class="u-embedded-media">
+                <iframe title="What is a MicroCloud?" class="u-embedded-media__element"
+                    src="https://www.youtube.com/embed/FnzUcGg60Qk" allowfullscreen></iframe>
```
Comment (immortalcodes) [inline templates/microcloud.html:110]: May be add alt description for the embedded video.

**Review by immortalcodes — CHANGES_REQUESTED** [review]

**Comment by immortalcodes** [comment]
Left some comments, good work!

File: run [hunk 4]
```diff
@@ -331,7 +331,7 @@ update_dependencies() {
         pip_dependencies_hash=$(docker run --volume ${etc_volume}:/etc ${dev_image} bash -c 'find $(find /usr/local/lib/ -maxdepth 1 -name "python*" -type d | sort | tail -n 1)/dist-packages -type f -print0 | sort -z | xargs -0 '${md5_command}' | '${md5_command}' | cut -c1-8')-${requirements_hash}
         if [ ! -f .pip.${project}.hash ] || [ "${pip_dependencies_hash}" != "$(cat .pip.${project}.hash)" ]; then
             echo "Installing new pip dependencies"
-            docker_run "" pip3 install --requirement requirements.txt
+            docker_run "" pip3 install --no-build-isolation --requirement requirements.txt
```
[the remark this answers is not in the document]
Comment (lizzochek) [reply run:334]: So apparently one of the python packages was removed: https://stackoverflow.com/questions/78806100/pip-install-randomwords-no-module-named-setuptools-command-test 
Therefore the ./run script wasn't working. I'm not sure about the wider effects, we can create a separate task to fix it

File: templates/microcloud.html (hunk 3 above)
Comment (lizzochek) [reply templates/microcloud.html:110]: I believe the title is enough here

**Comment by immortalcodes** [comment]
image
The orientation of image seems a bit of here, I understand `equal-hieght-row` class acts a bit strange.
One way could be not having the images on medium screen at all.

**Comment by lizzochek** [comment]
@immortalcodes let me check with @lyubomir-popov

**Comment by lyubomir-popov** [comment]
pls use the image utilities to set the background to highlight, and aspect ratio to 16:9

**Comment by immortalcodes** [comment]
LGTM! Good work.

**Review by immortalcodes — APPROVED** [review]

**Pull request merged**